vm/src/obj/objdict.rs

```diff
@@ -376,7 +370,7 @@ impl PyDict {
 
     #[pymethod]
     fn popitem(&self, vm: &VirtualMachine) -> PyResult {
-        if let Some((key, value)) = self.entries.pop_front() {
+        if let Some((key, value)) = self.entries.pop_back() {
```

Comment:
is this intended change?

Reply:
Yeah, it was a lot easier to have it pop from the back rather than the front, and it's actually supposed to be like that for dict (though not for set):

```py
>>> d = {}
>>> d[1] = 2
>>> d[2] = 3
>>> d[3] = 4
>>> d.popitem()
(3, 4)
>>> d.popitem()
(2, 3)
>>> d.popitem()
(1, 2)
>>> d.popitem()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
KeyError: 'popitem(): dictionary is empty'
>>> s = set()
>>> s.add(1)
>>> s.add(2)
>>> s.add(3)
>>> s.pop()
1
>>> s.pop()
2
>>> s.pop()
3
>>> s.pop()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
KeyError: 'pop from an empty set'
```
Hopefully it's an ok tradeoff for now, since dict having a correct behavior would probably be more depended on then set, and it didn't trip (or even fix) any tests

Reply:
interesting, I didn't know there was any difference about this